nixos/modules/services/system/nvme-rs.nix

```diff
@@ -0,0 +1,204 @@
+{
+  config,
+  options,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  inherit (lib) types;
+  cfg = config.services.nvme-rs;
+  opt = options.services.nvme-rs;
+  settingsFormat = pkgs.formats.toml { };
+in
+{
+  options.services.nvme-rs = {
+    enable = lib.mkEnableOption "nvme-rs, a monitoring service";
+
+    package = lib.mkPackageOption pkgs "nvme-rs" { };
+
+    settings = lib.mkOption {
+      type = types.submodule {
+        freeformType = settingsFormat.type;
+        options = {
```

Comment:
In the spirit of [RFC 0042](https://github.com/NixOS/rfcs/blob/master/rfcs/0042-config-option.md), it may be worth considering whether every setting here needs its own separate option. This is not a blocker, though; just some general advice for writing NixOS modules.

Reply:
Hi,

I preferred this approach because I find it clearer and more convenient for managing options separately.

Best Regards